ddtrace/contrib/pytest_bdd/plugin.py

```diff
@@ -72,23 +72,31 @@ def pytest_bdd_before_step(self, request, feature, scenario, step, step_func):
             span.set_tag(test.FRAMEWORK_VERSION, self.framework_version)
 
             # store parsed step arguments
-            parser = getattr(step_func, "parser", None)
-            if parser is not None:
-                converters = getattr(step_func, "converters", {})
-                parameters = {}
+            try:
+                parsers = [step_func.parser]
+            except AttributeError:
                 try:
-                    for arg, value in parser.parse_arguments(step.name).items():
-                        try:
-                            if arg in converters:
-                                value = converters[arg](value)
-                        except Exception:
-                            # Ignore invalid converters
-                            pass
-                        parameters[arg] = value
-                except Exception:
-                    pass
-                if parameters:
-                    span.set_tag(test.PARAMETERS, json.dumps(parameters))
+                    # pytest-bdd >= 6.0.0
+                    parsers = step_func._pytest_bdd_parsers
+                except AttributeError:
+                    parsers = []
+            for parser in parsers:
+                if parser is not None:
```

Comment:
🤔  this `if` is probably not required?

Reply:
it was there before, so probably better to keep it?

although, doing:

```python
for parser in parsers:
    if parser is None:
        continue
```

would remove the large indented block (this isn't really an issue/doesn't need to be changed)

Reply:
I think it was there because the `getattr` would return `None` if the attribute weren't there. But surely it doesn't hurt to do that check anyway I suppose 🤷